client/lib/application/src/application-detail/application-detail.jsx

```diff
@@ -44,6 +45,14 @@ export default function ApplicationDetail({applicationId, application, versions,
                         className='btn btn-primary'>
                         <Icon name='list' /> Versions
                     </Link>
+                    {ENV_DEVELOPMENT ?
+                        <Link
+                            to={Routes.lifecycle(LINK_PARAMS)}
+                            className='btn btn-primary'>
+                            <Icon name='list' /> Lifecycle
```

Comment:
Maybe [`bar-chart`](http://fontawesome.io/icon/bar-chart/) is a better icon than a list.